ast-checks/src/bin/unused-argument.rs

```diff
@@ -0,0 +1,62 @@
+use codespan::{FileId, Files};
+use nixpkgs_hammering_ast_checks::analysis::*;
+use nixpkgs_hammering_ast_checks::common_structs::*;
+use nixpkgs_hammering_ast_checks::tree_utils::walk_kind;
+use rnix::types::*;
+use rnix::SyntaxKind::*;
+use std::{env, error::Error};
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let args: Vec<String> = env::args().skip(1).collect();
+    println!("{}", analyze_files(args, analyze_single_file)?);
+    Ok(())
+}
+
+fn analyze_single_file(files: &Files<String>, file_id: FileId) -> Result<Report, Box<dyn Error>> {
+    let root = find_root(files, file_id)?;
+    let mut report: Report = vec![];
+
+    for lambda_elem in walk_kind(&root, NODE_LAMBDA) {
+        let lambda = lambda_elem.into_node().and_then(Lambda::cast);
+
+        let body = lambda
+            .clone()
+            .and_then(|l| l.body())
+            .ok_or("Unable to extract function body")?;
+        let identifiers_in_body: Vec<Ident> = walk_kind(&body, NODE_IDENT)
+            .filter_map(|elem| elem.into_node())
```

Comment:
Playing with [nix-explorer](https://gitlab.com/jD91mZM2/nix-explorer), maybe we could also filter those whose parent is `NODE_ATTRIBUTE` to eliminate (2):

![image](https://user-images.githubusercontent.com/705123/107586974-3848e300-6c01-11eb-8d96-bf5b9833e5c5.png)